pkg/routeagent_driver/handlers/ovn/utils.go

```diff
@@ -37,17 +37,24 @@ func getNextHopOnK8sMgmtIntf() (string, error) {
 	}
 
 	addrs, err := netLink.AddrList(link, k8snet.IPv4)
-	if err != nil || len(addrs) == 0 {
-		return "", errors.Wrapf(err, "failed to retrieve addresses for link")
+	if err == nil {
+		for _, addr := range addrs {
+			if addr.IPNet != nil {
+				return addr.IPNet.IP.String(), nil
+			}
+		}
 	}
 
-	for _, addr := range addrs {
-		if addr.IPNet != nil {
-			return addr.IPNet.IP.String(), nil
+	addrs, err = netLink.AddrList(link, k8snet.IPv6)
```

Comment:
As above, shouldn't we use the IP family specified for the handler?

Reply:
Done.